src/app/detail/ui/TaskEditor.tsx

```diff
@@ -152,23 +154,28 @@ export const TaskEditor = ({
 
       <Box mt="12px" sx={{ height: '100%' }}>
         <Tapwrite
-          uploadFn={async (file, tiptapEditorUtils) => {
-            const newBlob = await upload(file.name, file, {
-              access: 'public',
-              handleUploadUrl: '/api/upload',
-            })
-            tiptapEditorUtils.setImage(newBlob.url as string)
-          }}
           content={updateDetail}
           getContent={handleDetailChange}
           readonly={userType === UserType.CLIENT_USER}
           editorClass="tapwrite-details-page"
           placeholder="Add description..."
+          uploadFn={async (file, tiptapEditorUtils) => {
+            const supabaseActions = new SupabaseActions()
+            const fileName = generateRandomString(file.name)
+            const signedUrl: ISignedUrlUpload = await getSignedUrlUpload(fileName)
+            const filePayload = await supabaseActions.uploadAttachment(file, signedUrl, task_id)
+            const url = await supabaseActions.getPublicUrl(filePayload?.filePath ?? '')
+            tiptapEditorUtils.setImage(url, fileName)
+          }}
```

Comment:
I think we need to abstract these things out in a separate class or utility func. I understand that `supabaseActions` class already abstracts away, but the suggestion is that someone won't be able to understand what the each steps actually doing. So I think it's better to do something here?